### PR TITLE
Security perf tls02

### PR DESCRIPTION
**Pull request by pillip8282**

### CODE REVIEW

File: apps/examples/hello/ssl_client_tr.c [hunk 1]
```diff
@@ -0,0 +1,270 @@
+#include <tinyara/config.h>
```
Comment (jeongchanKim) [inline apps/examples/hello/ssl_client_tr.c:1]: Please add Samsung License.

File: apps/examples/hello/ssl_client_tr.c [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+			break;
+		}
+
+		len = ret;
+		mbedtls_printf(" %d bytes read\n\n%s", len, (char *)buf);
+	} while (1);
+
+	mbedtls_ssl_close_notify(&ssl);
+
+	exit_code = -1;
```
Comment (jeongchanKim) [inline apps/examples/hello/ssl_client_tr.c:249]: `exit_code` is initialized from line 64, but not used anywhere.
And it is set as `-1` again, but checked it is `-1` or not in line 254.
Please check it is needed or not.

File: apps/examples/performance/tls_handshake/tls_handshake_client.c [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+{
+	struct timespec ts;
+	SERVER_ADDR = ipaddr;
+	ts.tv_sec = 1612483200; // 2021-02-05 00:00:00
+	ts.tv_nsec = 0;
+
+	clock_settime(CLOCK_REALTIME, &ts);
+
+	int ret = 1, len;
+	int exit_code = -1;
```
Comment (jeongchanKim) [inline apps/examples/performance/tls_handshake/tls_handshake_client.c:62]: Please check `exit_code` is needed or not.

File: apps/examples/performance/tls_handshake/Kconfig [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+	bool "TLS handshake benchmark"
+	default n
+	depends on NET_SECURITY_TLS
+
+if EXAMPLES_TLS_HANDSHAKE
+
+config EXAMPLES_TLS_HANDSHAKE_PROGNAME
+	string "Program name"
+	default "tls_handshake"
+	depends on BUILD_KERNEL
```
Comment (jeongchanKim) [inline apps/examples/performance/tls_handshake/Kconfig:16]: I think this line is not needed.

File: external/mbedtls/alt/pk_wrap_alt.c [hunk 5]
```diff
@@ -190,10 +192,10 @@ static int rsa_verify_wrap(void *ctx,
 		free(pubkey.data);
 		return MBEDTLS_ERR_RSA_UNSUPPORTED_OPERATION;
 	}
-	ret = sl_set_key(shnd, key_type, key_idx, &pubkey, NULL, &hres);
+	ret = sl_set_key(shnd, key_type, key_idx, &pubkey, NULL);
 	free(pubkey_buf);
 	free(pubkey.data);
-	if (ret != SECLINK_OK || hres != HAL_SUCCESS) {
```
Comment (jeongchanKim) [inline external/mbedtls/alt/pk_wrap_alt.c:196]: Is it okay not to free the `pubkey_buf` and `pubkey.data`?

File: os/drivers/seclink/seclink_drv_utils.c [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+ 
+#include <tinyara/config.h>
```
Comment (jeongchanKim) [inline os/drivers/seclink/seclink_drv_utils.c:19]: I think most of the headers are not needed, even `tinyara/config.h`.
Please check them.

File: os/drivers/seclink/seclink_drv_utils.h [hunk 7]
```diff
@@ -1,34 +1,84 @@
...
+
+#define SLDRV_CALL(ret, res, method, param)                                         \
+	do {                                                                            \
+		if (se->ops->method) {                                                      \
+			sldrv_timer_handle hnd;                                                 \
+			(void)sldrv_start_time(&hnd);                                           \
+			res = (se->ops->method)param;                                           \
+			(void)sldrv_get_time(&hnd);                                             \
+			SLDRV_LOG(SLDRV_TAG "[PERF] T%d " #method " res %d elapsed %u %u ms\n", \
+					  getpid(), res, hnd.elapsed, hnd.elapsed / 1000);              \
```
Comment (jeongchanKim) [inline os/drivers/seclink/seclink_drv_utils.h:58]: For using `getpid()`, please include `unistd.h`.

File: apps/examples/performance/tls_handshake/tls_handshake_client.c [hunk 8]
```diff
@@ -0,0 +1,260 @@
+#include <tinyara/config.h>
```
Comment (jeongchanKim) [inline apps/examples/performance/tls_handshake/tls_handshake_client.c:1]: Please add Samsung License.

File: apps/examples/security_test/seclink/sl_crypto_rsa_test.c [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+
+#include <stdio.h>
+#include <string.h>
```
Comment (jeongchanKim) [inline apps/examples/security_test/seclink/sl_crypto_rsa_test.c:22]: Please remove unnecessary headers.
stdlib, pthread and so on.

File: apps/examples/security_test/seclink/sl_crypto_table.h [hunk 10]
```diff
@@ -0,0 +1,5 @@
+// command, type, handler
```
Comment (jeongchanKim) [inline apps/examples/security_test/seclink/sl_crypto_table.h:1]: Please add Samsung License.

File: apps/examples/security_test/seclink/sl_crypto_test.c [hunk 11]
```diff
@@ -0,0 +1,110 @@
...
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <pthread.h>
```
Comment (jeongchanKim) [inline apps/examples/security_test/seclink/sl_crypto_test.c:24]: I think `pthread.h` is not needed.

File: apps/examples/security_test/seclink/sl_test_table.h [hunk 12]
```diff
@@ -0,0 +1,4 @@
+SL_TEST_POOL("auth",   SL_TYPE_AUTH,   sl_handle_auth)
```
Comment (jeongchanKim) [inline apps/examples/security_test/seclink/sl_test_table.h:1]: Please add Samsung License.

File: apps/examples/security_test/seclink/sl_test_utils.c [hunk 13]
```diff
@@ -91,3 +91,32 @@ void sl_test_print_buffer(char *data, int buf_len, const char *message)
...
+		return -1;
+	}
+	opt->count = atoi(argv[3]);
+
+	if (strncmp(argv[2], "all", strlen("all") + 1) == 0) {
+		return tmax;
+	}
+
+	for (int i = 0; i < tmax; i++) {
+		if (strncmp(argv[2], command[i], strlen(command[i]) + 1) == 0) {
```
Comment (jeongchanKim) [inline apps/examples/security_test/seclink/sl_test_utils.c:117]: Please include `string.h` for `strncmp`.

File: external/mbedtls/alt/pk_wrap_alt.c [hunk 14]
```diff
@@ -827,88 +888,145 @@ int eckey_verify_wrap(void *ctx, mbedtls_md_type_t md_alg, const unsigned char *
...
-		return ret;
+	ret = sl_ecdsa_sign_md(shnd, ecdsa_mode, &t_hash, key_idx, &sign);
+	if (!is_injected_key) {
+		sl_remove_key(shnd, key_type, key_idx);
+	}
+	sl_deinit(shnd);
+	free(t_hash.data);
+
+	if (ret != SECLINK_OK) {
+		return MBEDTLS_ERR_ECP_HW_ACCEL_FAILED;
```
Comment (jeongchanKim) [inline external/mbedtls/alt/pk_wrap_alt.c:1024]: Is it okay not to free the `prikey.data`?

File: framework/src/security/security_auth.c [hunk 15]
```diff
@@ -274,15 +244,9 @@ security_error auth_verify_ecdsa_signature(security_handle hnd,
...
-
-	SECAPI_CALL(sl_ecdsa_verify_md(ctx->sl_hnd, hmode, &h_hash, &h_sign, key_idx, &hres));
-	if (hres != HAL_SUCCESS) {
-		SECAPI_HAL_RETURN(hres);
-	}
 
-
-	SECAPI_HAL_RETURN(hres);
+	SECAPI_CALL(sl_ecdsa_verify_md(ctx->sl_hnd, hmode, &h_hash, &h_sign, key_idx));
+SECAPI_RETURN(SECURITY_OK);
```
Comment (jeongchanKim) [inline framework/src/security/security_auth.c:249]: Please add the tab for indentation.

File: external/mbedtls/alt/pk_wrap_alt.c (hunk 14 above)
Comment (pillip8282) [reply external/mbedtls/alt/pk_wrap_alt.c:1024]: pirkey is freed at 1008 line

**Review by jeongchanKim — APPROVED** [review]
LGTM

File: apps/examples/performance/tls_handshake/tls_handshake_client.c [hunk 16]
```diff
@@ -0,0 +1,277 @@
...
+	mbedtls_ssl_config conf;
+	mbedtls_x509_crt cacert;
+	int ret = 1;
+	int len = 0;
+	struct timespec ts;
+	SERVER_ADDR = ipaddr;
+	ts.tv_sec = 1612483200; // 2021-02-05 00:00:00
+	ts.tv_nsec = 0;
+
+	clock_settime(CLOCK_REALTIME, &ts);
```
Comment (an4967) [inline apps/examples/performance/tls_handshake/tls_handshake_client.c:88]: The time is set '2021-02-05 00:00:00' at here, but the certificates are,
```
Validity
Not Before: Aug 13 04:23:30 2021 GMT
Not After : Dec 31 04:23:30 2069 GMT
```

I think the date should be set 'Sep / Oct' for test

**Review by an4967 — APPROVED** [review]
LGTM.
It works well

**Pull request merged**